configs/configupgrade/upgrade_expr.go

```diff
@@ -38,9 +38,9 @@ Value:
 		return upgradeExpr(tv.Token, filename, interp, an)
 
 	case hcl1token.Token:
-		litVal := tv.Value()
```

Comment:
It's unclear from the diff why this was moved to happen in the switch cases vs leaving it here?

Reply:
`Value()`  panics if there is an error converting the token Text value to the indicated type (https://github.com/hashicorp/hcl/blob/master/hcl/token/token.go#L128). 

In normal terraform usage, some variation of `validate()` would have caught that the token literal value was too long and exited with an error before now. Since that's not happening in `0.12upgrade`, I moved the `Value()` call into the individual statements so that I could skip it altogether when the token is a number and instead directly convert the `Text` to int or float (or not, in the case of an error) without panicking.

Reply:
ahhhh thank you! TIL